### DISCUSSION TITLE
2FA without other device

### DISCUSSION THREAD

**Discussion by peepo** [community/community#139622] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

How can I enable 2FA without a mobile phone, SMS or similar.
I do not use any mobile phone device at all.
Thank you

**Comment by davevad93** [comment]
Hi @peepo ,

In the two-factor methods settings, under the Authenticator App option, it's indicated that you can also use 2FA with a TOTP browser extension that generates the codes directly in your browser:

![00000](https://github.com/user-attachments/assets/889dd25c-0548-4c6e-bf77-fc7bdd9470b8)

Personally I don't use any browser extension as 2FA so I can't recommend you any specific one, but if you search the web you'll surely find something.

**Reply by peepo** [reply]
Thanks,
iirc the way I read it, one needs a mobile app in the first instance.
Did I misread that?

**Reply by davevad93** [reply]
Hey @peepo you're welcome. I don't think so, mobile is required for authentication via SMS and the GitHub App but the Authenticator option should be for both mobile and pc devices since there are various browser extensions that can be used as 2FA. There's  this one for example, [Authenticator](https://github.com/Authenticator-Extension/Authenticator), but I've never used it and I'm not sure if it works with GitHub.

**Reply by davevad93** [reply]
I forgot that there's also the [Passkeys](https://docs.github.com/en/authentication/authenticating-with-a-passkey/about-passkeys) option that satisfies both password and 2FA requirements, and can be used with `Windows Hello` if you're a Windows user.

**Comment by peepo** [comment]
the browser related part appears to be:
[Configuring two-factor authentication using a security key](https://docs.github.com/en/enterprise-server@3.10/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-security-key)

which states:
You must have already configured 2FA via a TOTP mobile app.

**Reply by davevad93** [reply]
I see @peepo , but I've read from some GitHub users who claimed to be able to set up a 2FA method without mobile phone/app, so I thought it was possible, e.g. take a look at [this dicussion](https://github.com/orgs/community/discussions/66191#discussioncomment-7084989), or this [post](https://www.reddit.com/r/github/comments/15z9y1r/is_there_a_way_to_use_github_2fa_without_using_a/?rdt=46641).

I honestly don't think it's fair that a user is forced to have a mobile phone in order to configure 2FA, there has to be some alternative.

**Comment by formbi-kupiec** [comment]
You can use KeepassXC. It doesn't always work, but it's on the Github side (very ironic since they're enforcing it the most), works perfectly fine everywhere else.
